### DISCUSSION TITLE
Custom plugin failed to modify the content-length

### DISCUSSION THREAD

**Discussion by zoudlgit** [Kong/kong#10735] [Help] [answered]
I want to modify the body and content-length of the response returned, where the modification of the body is effective but the modification of the content-length is not effective.

**Comment by zoudlgit** [comment]


**Reply by ms2008** [reply, marked as answer]
From https://github.com/openresty/lua-nginx-module#body_filter_by_lua_block

> When the Lua code may change the length of the response body, then it is required to always clear out the `Content-Length` response header (if any) in a header filter to enforce streaming output,
